File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
       int lp=1,rp=1;
       int []ans=new int[nums.length];
       for(int i=0;i<nums.length;i++){
        ans[i]=lp;
        lp*=nums[i];
       } 
       for(int i=nums.length-1;i>=0;i--){
        ans[i]*=rp;
        rp*=nums[i];
       }
       return ans;
    }
}  

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        if(nums.size()==1)return nums[0];
        if(nums.size()==0)return 0;
        int l=0,r=nums.size()-1;
        int min_f=INT_MAX;
        while(l<r){
            if(nums[l]<nums[r]){
                min_f=min(min_f,nums[l]);
                r--;
            }
            else{
                min_f=min(min_f,nums[r]);
                l++;
            }
        }
        return min_f;
    }
};
